climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1387 @@
+                (https://firms.modaps.eosdis.nasa.gov/download/)
+            centr_res_factor : float, optional, default=1
+                resolution factor with respect to the satellite data to use
+                for centroids creation. Hence, if MODIS data (1 km res) is
+                used and centr_res_factor is set to 0.2, the grid spacing of
+                the generated centroids will equal 5 km (=1/0.2). If centroids
+                are defined, this parameter has no effect.
+            centroids : Centroids, optional
+                centroids in degrees to map data
+
+        Returns
+        -------
+            self : climada.hazard.WildFire instance
+
+        """
+        self.clear()
+
+        # read and initialize data
+        firms = self._clean_firms_csv(csv_firms)
+        # compute centroids
+        res_data = self._firms_resolution(firms)
+        if not centroids:
+            centroids = self._firms_centroids_creation(firms, res_data, centr_res_factor)
+        else:
+            if not centroids.lat.any():
+                centroids.set_meta_to_lat_lon()
+        res_centr = self._centroids_resolution(centroids)
+
+        # fire identification
+        while firms.iter_ev.any():
+            # Compute cons_id: consecutive fires in current iteration
+            self._firms_cons_days(firms)
+            # Compute clus_id: cluster identifier inside cons_id
+            self._firms_clustering(firms, res_data, self.clus_thres_firms)
+            # compute event_id
+            self._firms_fire(self.days_thres_firms, firms.cons_id.values, \
+                             firms.clus_id.values, firms.event_id.values, \
+                             firms.iter_ev.values, firms.datenum.values)
+            LOGGER.info('Remaining fires to identify: %s.', str(np.argwhere(\
+            firms.iter_ev.values).size))
+
+        # remove minor fires
+        if self.remove_minor_fires_firms:
+            firms = self._firms_remove_minor_fires(firms,
+                                                   self.minor_fire_thres_firms)
+
+        # compute brightness and fill class attributes
+        LOGGER.info('Computing intensity of %s fires.',
+                    np.unique(firms.event_id).size)
+        self._calc_brightness(firms, centroids, res_centr)
+
+
+    def hull_burned_area(self, ev_id, alpha=100.87, return_plot=False):
+        """Compute the burned area for a given fire.
+
+        Please note: NASA advises against calculating burned area using the
+        FIRMS data. See: https://earthdata.nasa.gov/faq/firms-faq
+
+        Algorithm used: https://pypi.org/project/alphashape/
+
+        Parameters
+        ----------
+            ev_id : int
+                id of the selected fire
+            alpha : float, optional, default=100.87
+                parameter used to compute the concave hull. Default is suited
+                for 1 km MODIS data and calibrated over California
+                (no reference)
```

Comment:
Then please explain what this number means. What are the units? What is its meaning?

Reply:
removed that part